yt_dlp/extractor/bilibili.py

```diff
@@ -535,6 +539,80 @@ def _real_extract(self, url):
         return self.playlist_result(self._entries(list_id), list_id)
 
 
+# https://www.bilibili.com/v/kichiku/mad/#/
+class BilibiliCategoryIE(InfoExtractor):
+    IE_NAME = 'Bilibili category extractor'
+    _MAX_RESULTS = 1000000
+    _VALID_URL = r'https?://www\.bilibili\.com/v/[a-zA-Z]+\/[a-zA-Z]+'
+    _TESTS = [{
+        'url': 'https://www.bilibili.com/v/kichiku/mad',
+        'info_dict': {
+            'id': 'kichiku: mad',
+            'title': 'kichiku: mad'
+        },
+        'playlist_mincount': 45,
+        'params': {
+            'playlistend': 45
+        }
+    }]
+
+    def _fetch_page(self, api_url, num_pages, query, page_num):
+        parsed_json = self._download_json(
+            api_url + '&pn=%s' % page_num, query, query={'Search_key': query},
+            note='Extracting results from page %s of %s' % (page_num, num_pages))
+
+        video_list = try_get(parsed_json, lambda x: x['data']['archives'], list)
+        if not video_list:
+            raise ExtractorError('failed to retrieve video list for page %d' % page_num)
```

Comment:
Shouldn't it be like this?
```suggestion
            raise ExtractorError('Failed to retrieve video list for page %d' % page_num)
```